core/dbt/include/global_project/macros/adapters/common.sql

```diff
@@ -81,16 +97,130 @@
+{% endmacro %}
+
+{% macro default__alter_column_type(relation, column_name, new_column_type) -%}
+  {#
+    1. Create a new column (w/ temp name and correct type)
+    2. Copy data over to it
+    3. Drop the existing column (cascade!)
+    4. Rename the new column to existing column
+  #}
+  {%- set tmp_column = column_name + "__dbt_alter" -%}
+
+  {% call statement('alter_column_type') %}
+    alter table {{ relation }} add column {{ tmp_column }} {{ new_column_type }};
+    update {{ relation }} set {{ tmp_column }} = {{ column_name }};
+    alter table {{ relation }} drop column {{ column_name }} cascade;
+    alter table {{ relation }} rename column {{ tmp_column }} to {{ column_name }}
+  {% endcall %}
+
+{% endmacro %}
+
+
+{% macro drop_relation(relation) -%}
+  {{ return(adapter_macro('drop_relation', relation)) }}
+{% endmacro %}
+
+
+{% macro default__drop_relation(relation) -%}
+  {% call statement('drop_relation', auto_begin=False) -%}
+    drop {{ relation.type }} if exists {{ relation }} cascade
+  {%- endcall %}
+{% endmacro %}
+
+{% macro truncate_relation(relation) -%}
+  {{ return(adapter_macro('truncate_relation', relation)) }}
 {% endmacro %}
+
+
+{% macro default__truncate_relation(relation) -%}
+  {% call statement('truncate_relation') -%}
+    truncate table {{ relation }}
+  {%- endcall %}
+{% endmacro %}
+
+{% macro rename_relation(from_relation, to_relation) -%}
+  {{ return(adapter_macro('rename_relation', from_relation, to_relation)) }}
+{% endmacro %}
+
+{% macro default__rename_relation(from_relation, to_relation) -%}
+  {% set target_name = adapter.quote_as_configured(to_relation.identifier, 'identifier') %}
+  {% call statement('rename_relation') -%}
+    alter table {{ from_relation }} rename to {{ target_name }}
+  {%- endcall %}
+{% endmacro %}
+
+
+{% macro information_schema_name(database) %}
+  {{ return(adapter_macro('information_schema_name', database)) }}
+{% endmacro %}
+
+{% macro default__information_schema_name(database) -%}
+  {%- if database -%}
+    "{{ database }}".information_schema
+  {%- else -%}
+    information_schema
+  {%- endif -%}
+{%- endmacro %}
+
+
+{% macro list_schemas(database) -%}
+  {{ return(adapter_macro('list_schemas', database)) }}
+{% endmacro %}
+
+{% macro default__list_schemas(database) -%}
+  {% call statement('list_schemas', fetch_result=True, auto_begin=False) %}
+    select distinct schema_name
+    from {{ information_schema_name(database) }}.schemata
+    where catalog_name='{{ database }}'
+  {% endcall %}
+  {{ return(load_result('list_schemas').table) }}
+{% endmacro %}
+
+
+{% macro check_schema_exists(database, schema) -%}
+  {{ return(adapter_macro('check_schema_exists', database)) }}
+{% endmacro %}
+
+{% macro default__check_schema_exists(database, schema) -%}
+  {% call statement('check_schema_exists', fetch_result=True, auto_begin=False) -%}
+        select count(*)
+        from {{ information_schema_name(database) }}.schemata
```

Comment:
i mentioned this in #1202 but these `information_schema` queries are worth discussing. Just flagging this so we don't miss it

Reply:
I believe I've resolved these issues by implementing postgres/redshift versions that query pg_*. The catalog still uses information_schema, but it always has - is that something to change? The permissions implications aren't totally clear to me on that front.

Reply:
huh, that's interesting. I think we do get more info from the information_schema tables, so I think it's probably right to continue using them here.

The catalog is solely used for docs (correct me if i'm wrong), so I guess that's ok. If a user doesn't have read access to a table, then it just won't appear in the documentation. That's not the most ideal scenario, and it's not something I think we've done intentionally, but it's also pretty reasonable IMO.

The bigger problem with information_schema queries is that dbt might fail to see a table, then try to create it rather than eg. dropping it first. The user will get an error message either way, but the `drop` error will be an obvious permissions-related error, whereas the "this already exists" error will look like a cache inconsistency problem.

Reply:
There's always this stack exchange post, but I would really rather not use it: https://dba.stackexchange.com/a/195721